MockKitUI/MockSupport.swift

```diff
@@ -0,0 +1,144 @@
+    }
+    
+    public func setAlertIssuer(alertIssuer: AlertIssuer?) {
+        self.alertIssuer = alertIssuer
+    }
+    
+    public func supportMenuItem(supportInfoProvider: SupportInfoProvider, urlHandler: @escaping (URL) -> Void) -> AnyView? {
+        return AnyView(SupportMenuItem(mockSupport: self))
+    }
+    
+    public func softwareUpdateView(guidanceColors: GuidanceColors, bundleIdentifier: String, currentVersion: String, openAppStoreHook: (() -> Void)?) -> AnyView? {
+        return AnyView(
+            Button("versionUpdate: \(versionUpdate!.localizedDescription)\n\nbundleIdentifier: \(bundleIdentifier)\n\ncurrentVersion: \(currentVersion)") {
+                openAppStoreHook?()
+            }
+        )
+    }
+}
+
+extension MockSupport {
+    
+    var alertCadence: TimeInterval {
+        return TimeInterval.minutes(1)
+    }
+
+    private func maybeIssueAlert(_ versionUpdate: VersionUpdate) {
+        guard versionUpdate >= .recommended else {
+            noAlertNecessary()
+            return
+        }
+        
+        let alertIdentifier = Alert.Identifier(managerIdentifier: MockSupport.supportIdentifier, alertIdentifier: versionUpdate.rawValue)
+        let alertContent: LoopKit.Alert.Content
+        if firstAlert {
+            alertContent = Alert.Content(title: versionUpdate.localizedDescription,
+                                         body: NSLocalizedString("""
+                                                Your app is out of date. It will continue to work, but we recommend updating to the latest version.
+                                                """, comment: "Alert content body for first software update alert"),
+                                         acknowledgeActionButtonLabel: NSLocalizedString("OK", comment: "default acknowledgement"),
+                                         isCritical: versionUpdate == .required)
+        } else if let lastVersionCheckAlertDate = lastVersionCheckAlertDate,
+                  abs(lastVersionCheckAlertDate.timeIntervalSinceNow) > alertCadence {
+            alertContent = Alert.Content(title: NSLocalizedString("Update Reminder", comment: "Recurring software update alert title"),
+                                         body: NSLocalizedString("""
+                                                Your app is still out of date. It will continue to work, but we recommend updating to the latest version.
+                                                """, comment: "Alert content body for recurring software update alert"),
+                                         acknowledgeActionButtonLabel: NSLocalizedString("OK", comment: "default acknowledgement"),
+                                         isCritical: versionUpdate == .required)
+        } else {
+            return
+        }
+        alertIssuer!.issueAlert(Alert(identifier: alertIdentifier, foregroundContent: alertContent, backgroundContent: alertContent, trigger: .immediate))
```

Comment:
If `alertIssuer` is optional (see `setAlertIssuer(alertIssuer: AlertIssuer?)`), then should we really force it here? Either make `alertIssuer` required or make this `alertIssuer?`.